src/lib.rs

```diff
@@ -1197,6 +1458,33 @@ safety_comment! {
     ///
     /// [1] https://doc.rust-lang.org/reference/types/textual.html
     unsafe_impl!(char: FromZeroes, AsBytes);
+    /// SAFETY:
+    /// - `MaybeUninit<char>` has no bit validity requirements, and has the same
+    ///   alignment as `char`. Since the `is_bit_valid` caller must promise that
+    ///   the provided `NonNull<char>` is properly-aligned and points a region
+    ///   which is valid for reads, that's all we need to ensure that converting
+    ///   it to a `&MaybeUninit<char>` is sound.
+    /// - Since we transmute `c` from the bytes passed to `is_bit_valid` without
+    ///   modifying them, and since `char::from_u32` guarantees that it returns
+    ///   `None` if its input is not a valid `char` [1], this function only
+    ///   returns `true` if its argument is a valid `char`, and so this is a
+    ///   sound implementation of `TryFromBytes::is_bit_valid`.
+    ///
+    /// Note that it might be slightly simpler to treat `candidate` as a `[u8;
+    /// 4]` - it would make the code and the safety argument a bit simpler.
+    /// However, using `&MaybeUninit<char>` ensures that the compiler preserves
+    /// alignment information, which it may be able to use to produce smaller or
+    /// better-performing assembly.
+    ///
+    /// [1] https://doc.rust-lang.org/std/primitive.char.html#method.from_u32
+    unsafe_impl!(char: TryFromBytes; |candidate: &MaybeUninit<char>| {
```

Comment:
Leave a TODO that, once `char` is guaranteed to have the same alignment as `u32`, we can use that instead of `MaybeUninit<char>`.